### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by NasKhalil**
This branch includes the following featured implemented: 

- Create a class `Corrector` with a method `correct_name`
- Set an instance of `Corrector` in `Person` on initialization.
- In `Person` add a method `validate_name` that uses the `Corrector` instance to validate and save the corrected name.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require './corrector'
+
 class Person
   attr_accessor :name, :age
   attr_reader :id
```
Comment (andersonlebon) [inline person.rb]: - [x] kindly make sure the Person class inherits from the Namable class. also, It should have a method `correct_name`. :point_down: 

```ruby
class Person < Nameable
  .
  .

    def correct_name
       name
    end
  .
  .
end
```

File: corrector.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Corrector
+  def correct_name(name)
+    corrected_name = name.capitalize
+    corrected_name = corrected_name[0..9] if name.length > 10
+    corrected_name
+  end
+end
```
Comment (andersonlebon) [inline corrector.rb]: - [x] Kindly note that you need to follow very well the [requirement of this project ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md) by :point_down: 
- Creating `Namable` class with a method 'correct_name' raising an error. 
-  Checking if the `Decorator` class inherits from Namable.
-  Checking if in the constructor a namable object from params is assigned to an instance variable.
-  Checking of the correct_name method that returns the result of the correct_name method of the @namable is implemented.
- Creating a class that inherits from the base Decorator class.
- Implementing a method correct_name that capitalizes the output of @nameable.correct_name.
- Creating a class that inherits from the base Decorator class.
- Implementing a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

#### Kindly make sure everything is done in the :point_right: [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md) before requesting the next review. Thanks.

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @NasKhalil 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [x] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kingstalux — APPROVED** [review]
Hi @NasKhalil,
# Status: Approved ✔️ ✔️ ✔️ 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**